iModelCore/ECDb/ECDb/ECSchemas/ECDbMeta.04.00.02.ecschema.xml

```diff
@@ -0,0 +1,669 @@
+        <ECProperty propertyName="ContainerType"  typeName="int"/>
+        <ECProperty propertyName="Ordinal"  typeName="int"/>
+        <ECProperty propertyName="Instance"  typeName="string"/>
+        <ECNavigationProperty propertyName="Class" relationshipName="CustomAttributeClassHasInstance" direction="backward"/>
+    </ECEntityClass>
+    <ECRelationshipClass typeName="CustomAttributeClassHasInstance" modifier="Sealed" strength="referencing">
+        <Source multiplicity="(1..1)" roleLabel="has instance" polymorphic="false">
+            <Class class="ECClassDef"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="is instance of" polymorphic="false">
+            <Class class="CustomAttribute"/>
+        </Target>
+    </ECRelationshipClass>
+
+    <!-- Concrete Custom Attribute instances for property/class/schema -->
+    <ECEntityClass typeName="PropertyCustomAttribute" description="" displayLabel="" modifier="Abstract">
+        <ECCustomAttributes>
+            <View xmlns="ECDbMap.02.00.03">
+                <Query>
+                    SELECT
+                        [ca].[ECInstanceId],
+                        NAVIGATION_VALUE(meta.PropertyCustomAttribute.Property, [ca].[ContainerId]),
+                        NAVIGATION_VALUE(meta.PropertyCustomAttribute.CustomAttributeClass, [ca].[Class].[Id]),
+                        XmlCAToJson([ca].[Class].[Id], [ca].[Instance]) [ValueJson]
+                    FROM [meta].[CustomAttribute] [ca]
+                    WHERE [ca].[ContainerType] = 992
+                    ORDER BY [ca].[Ordinal]
+                </Query>
+            </View>
+        </ECCustomAttributes>
+        <ECNavigationProperty propertyName="Property" relationshipName="PropertyHasCustomAttribute" direction="backward"/>
+        <ECNavigationProperty propertyName="CustomAttributeClass" relationshipName="CustomAttributeClassHasInstanceOnProperty" direction="backward"/>
+        <ECProperty propertyName="ValueJson"  typeName="string"/>
+    </ECEntityClass>
+    <ECRelationshipClass typeName="PropertyHasCustomAttribute" modifier="Abstract" strength="referencing">
+        <Source multiplicity="(1..1)" roleLabel="has custom attribute" polymorphic="false">
+            <Class class="ECPropertyDef"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="is set on" polymorphic="false">
+            <Class class="PropertyCustomAttribute"/>
+        </Target>
+    </ECRelationshipClass>
+    <ECRelationshipClass typeName="CustomAttributeClassHasInstanceOnProperty" modifier="Abstract" strength="referencing">
+        <Source multiplicity="(1..1)" roleLabel="has instance on" polymorphic="false">
+            <Class class="ECClassDef"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="is instance of" polymorphic="false">
+            <Class class="PropertyCustomAttribute"/>
+        </Target>
+    </ECRelationshipClass>

```

Comment:
It's weird that these relationships are abstract but there is no other indication that they are 'virtual' or 'view' relationships.  The view classes at least have a view CA so it is more obvious

Reply:
Hm yeah. Agreed. I'm not sure what to do about it. It is kind of a shortcut hack that Simonas came up with, because they can get all their information by looking for a navigation property. So giving them a view definition would be redundant.

Should we add a new custom attribute for these types of relationships? Like <DynamicallyCreatedViewBasedOnForeignKey>? I know it's a silly name, we need a better one, it's just to get the idea across...

Reply:
Added a CA to indicate the behavior